app/Http/Controllers/Webapp/PostController.php

```diff
@@ -0,0 +1,1890 @@
+        if(!$this->currentUser->is(User::ADMIN)) {
+            $userQueryWhere = ' AND is_delete = 0 ';
+        }
+        if($this->currentUser->is(User::EDITOR) || $this->currentUser->is(User::CONTRIBUTOR)) {
+            $userQueryWhere .= ' AND p.user_id = '.$this->currentUser->id;
+        }
+        if($request->get('keyword')) {
+            $_kw = urldecode($request->get('keyword'));
+            $queryWhere = '((p.site_id > 0 AND s.id > 0) OR (p.site_id = 0 OR p.site_id IS NULL)) AND keyword=\'' . mysqli_real_escape_string(new \mysqli(env('DB_HOST'), env('DB_USERNAME'), env('DB_PASSWORD'), env('DB_DATABASE'), env('DB_PORT')), $_kw) . '\' AND k.type = 1'.$userQueryWhere;
+        } else if(empty($request->all())) {
+            $queryWhere = '((p.site_id > 0 AND s.id > 0) OR (p.site_id = 0 OR p.site_id IS NULL)) ' . $userQueryWhere . ' GROUP BY p.id';
+        } else {
+            $filters = array_column($request->all(), 'filters');
+
+            if($filters && count($filters[0]) == 1 && isset($filters[0]['records_per_page'])) {
+                $queryWhere = '((p.site_id > 0 AND s.id > 0) OR (p.site_id = 0 OR p.site_id IS NULL)) '.$userQueryWhere . ' GROUP BY p.id';
+            } else if(count($filters) == 0) {
+                $queryWhere = '((p.site_id > 0 AND s.id > 0) OR (p.site_id = 0 OR p.site_id IS NULL)) '.$userQueryWhere . ' GROUP BY p.id';
+            } else {
+                $queryWhere = '((p.site_id > 0 AND s.id > 0) OR (p.site_id = 0 OR p.site_id IS NULL)) '.$userQueryWhere;
+            }
+
+        }
+
+        $query->where($queryWhere);
+
+        $cfg = (new GridConfig())
+          ->setDataProvider(
+            new DbalDataProvider($query)
+          )
+          ->setColumns([
+            (new FieldConfig)
+              ->setName('id')
+              ->setLabel('ID')
+              ->addFilter(
+                (new SelectFilterConfig())
+                  ->setOperator(FilterConfig::OPERATOR_EQ)
+                  ->setDefaultValue(null)
+                  ->setTemplate('webapp.components.nayjest_columns_filter')
+                  ->setFilteringFunc(function($val, DbalDataProvider $dap) {
+                    return $dap->getBuilder()->andWhere("p.id = $val")->addGroupBy('p.id');
+                })
+              )
+              ->setSortable(true)
+              ->setSorting(Grid::SORT_DESC)
+              ->setCallback(function($pid) {
+                  return "<input type='checkbox' name='selected_ids[]' value='$pid' style='display: none'/>".$pid;
+              }),
+            (new FieldConfig)
+              ->setName('title')
+              ->setLabel('タイトル')
+              ->setSortable(true)
+              ->setSorting(Grid::SORT_ASC)
+              ->setCallback(function ($val, ObjectDataRow $row) {
+                  $post_id = $row->getCellValue('id');
+                  return '<a href="' . action('Webapp\PostController@getEdit', $post_id) . '"</a>'.$val.'</td>';
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "<a href="" 7 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.relipa.blog/coding_rules#rule_key=php%3AS1192)